tests/Node.py

```diff
@@ -326,21 +327,19 @@ def isBlockFinalized(self, blockNum):
         return self.isBlockPresent(blockNum, blockType=BlockType.lib)
 
     class BlockWalker:
-        def __init__(self, node, transOrTransId, startBlockNum=None, endBlockNum=None):
-            assert(isinstance(transOrTransId, (str,dict)))
-            if isinstance(transOrTransId, str):
-                self.trans=None
-                self.transId=transOrTransId
-            else:
-                self.trans=transOrTransId
-                self.transId=Node.getTransId(trans)
+        def __init__(self, node, transId, startBlockNum=None, endBlockNum=None):
+            assert(isinstance(transId, str))
+            self.trans=None
+            self.transId=transId
             self.node=node
             self.startBlockNum=startBlockNum
             self.endBlockNum=endBlockNum
 
         def walkBlocks(self):
             start=None
             end=None
+            if self.trans is None and self.transId in self.transCache:
+                self.trans=self.transCache[self.transId]
```

Comment:
This will throw a `KeyError` exception if `self.transId` is not in `self.transCache`.  It will not return `None`.

Reply:
I need self.transId in self.transCache.keys(), right?

Reply:
Use `self.transCache.get(self.transId)`.  That returns `None` if the key is not present in the dictionary.